src/tests/ftest/cart/util/cart_logtest.py

```diff
@@ -528,10 +528,16 @@ def _check_pid_from_log_file(self,
                             show_line(line, 'HIGH', 'free of unknown memory')
                         err_count += 1
                 elif line.is_realloc():
-                    new_pointer = line.get_field(-3)
-                    old_pointer = line.get_field(-1)[:-2].split(':')[-1]
+                    new_pointer = line.get_field(-6)
+                    old_pointer = line.get_field(-1)[0:-2]
+                    old_sz = int(line.get_field(-3))
+                    new_sz = line.calloc_size()
                     if new_pointer != '(nil)' and old_pointer != '(nil)':
-                        memsize.subtract(regions[old_pointer].calloc_size())
+                        exp_sz = regions[old_pointer].calloc_size()
+                        if old_sz not in [0, exp_sz, new_sz]:
```

Comment:
I think this is preferred, but it's minor.
```suggestion
                        if old_sz not in (0, exp_sz, new_sz):
```

Reply:
Ok, decided to land it as is